File: src/main/java/sigmaone/industrialism/energy/IEnergyHandler.java
package sigmaone.industrialism.energy;

public interface IEnergyHandler {
    /*
     * Inserts energy into the handler
     *
     * @param amount How much energy to insert
     */
    float putEnergy(float amount);
    /*
     * Extracts energy from the handler
     *
     * @param amount How much energy to extract
     *
     * @return Amount of energy extracted
     */
    float takeEnergy(float amount);
}

File: src/main/kotlin/sigmaone/industrialism/energy/IEnergyHandler.kt
package sigmaone.industrialism.energy

interface IEnergyHandler {
    /*
     * Inserts energy into the handler
     *
     * @param amount How much energy to insert
     */
    fun putEnergy(amount: Float): Float

    /*
     * Extracts energy from the handler
     *
     * @param amount How much energy to extract
     *
     * @return Amount of energy extracted
     */
    fun takeEnergy(amount: Float): Float
}